windows/ReactNativeAsyncStorage/RNCAsyncStorage.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include "pch.h"
+#include "NativeModules.h"
+#include "DBStorage.h"
+
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Data::Json;
+
+namespace ReactNativeAsyncStorage
+{
+    REACT_MODULE(RNCAsyncStorage);
+    struct RNCAsyncStorage
+    {
+        DBStorage dbStorage;
+
+        REACT_METHOD(multiGet);
+        void multiGet(std::vector<JSValue>& keys, std::function<void(JSValueArray const& errors, JSValueArray const& results)>&& callback) noexcept {
+            dbStorage.AddTask(DBStorage::DBTask::Type::multiGet, std::move(keys),
+                [callback](std::vector<JSValue> const& callbackParams) {
+                    if (callbackParams.size() > 0) {
+                        auto& errors = callbackParams[0].AsArray();
+                        if (callbackParams.size() > 1) {
+                            callback(errors, callbackParams[1].AsArray());
+                        }
+                        else {
+                            callback(errors, {});
+                        }
+                    }
+                });
+        }
+
+        REACT_METHOD(multiSet);
+        void multiSet(std::vector<JSValue>& pairs, std::function<void(JSValueArray const&)>&& callback) noexcept {
+            dbStorage.AddTask(DBStorage::DBTask::Type::multiSet, std::move(pairs),
+                [callback](std::vector<JSValue> const& callbackParams) {
```

Comment:
Move `callback` into the closure?

Reply:
You mean make callback a parameter of the lambda?

I wasn't able to since callback types are different. `CxxModule::Callback` is not available here. If you look closely, some callbacks are `std::function<void(JSValueArray const&)>`, others are `std::function<void(JSValueArray const& errors, JSValueArray const& results)>`. We can make `AddTask` take different parameters but then we moved the burden to `DBTask`, it has to take account all the different callback types, which feels more complicated than just capturing the callback here.

Reply:
No, I mean something like, say:
```c++
[cb{std::move(callback)}](std::vector<JSValue> const& callbackParams) {
  ...
  cb(...);
  ...
}
```

Reply:
Oh I see, will update. Thanks for the example!